### ISSUE TITLE
Carthage not working for version 0.8.2 and beyond

### ISSUE THREAD

**Issue by WilliamIzzo83** [grpc/grpc-swift#449]
### New Issue Checklist

- [ ] Updated grpc-swift to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/grpc/grpc-swift/blob/master/CONTRIBUTING.md)
- [x] I searched for [existing GitHub issues](https://github.com/grpc/grpc-swift/issues)(especially, there are many [resolved issues related to SSL](https://github.com/grpc/grpc-swift/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+ssl))

### Issue Description

Hi!

I get the following linker error when trying to get grpc-swift through carthage

```
ld: warning: directory not found for option '-L/usr/local/Cellar/nghttp2/1.37.0/lib'
ld: library not found for -lnghttp2
```

It seems that nghttp2 library can't be found. I tried to get it with mac ports first and the error is still there. Then by looking at the flag ```-L/usr/local/Cellar/nghttp2/1.37.0/lib``` I thought that maybe with brew I could do it, but I had no luck because brew will install 1.38.0

Any ideas?

**Comment by MrMage** [comment]
I think Homebrew has an option to install older versions; did you try that?

**Comment by WilliamIzzo83** [comment]
Yes I was looking into it and it seems quite [complex](https://zoltanaltfatter.com/2017/09/07/Install-a-specific-version-of-formula-with-homebrew/), not to mention that a simple brew update will break it again. Also in my case I had to install specifically brew: usually I can do everything with macports.

I was thinking if there could be other options, because this setup seems really brittle. Of course I'm willing to contribute.

**Comment by MrMage** [comment]
That makes sense. Thinking about this more, I think the dependency on `nghttp2` is only an artifact of the Carthage project being generated by SwiftPM — our CocoaPods variant does not require `nghttp2`, after all. I think it should be possible to modify https://github.com/grpc/grpc-swift/blob/master/patch-carthage-project.rb to remove this dependency from the resulting project.

**Comment by WilliamIzzo83** [comment]
All right, I can give this a shot

**Comment by MrMage** [comment]
Fixed by #450.
